### PR TITLE
Fix TS_EVENT_VCONN_EOS crash

### PR DESCRIPTION
**Pull request by mraerino**
Related to https://github.com/apache/trafficserver/issues/5577

Fixes https://github.com/netlify/traffic-mesh/issues/827

### CODE REVIEW

**Review by calavera — APPROVED** [review]

**Review by calavera — APPROVED** [review]
that makes sense

File: proxy/ProxyClientSession.cc [hunk 1]
```diff
@@ -162,13 +162,9 @@ ProxyClientSession::state_api_callout(int event, void *data)
...
   // this is a condition that sometimes
   // happens and causes crashes
-  case TS_EVENT_VCONN_EOS:
+  default:
     TSError("WARN: Invalid event received in ProxyClientSession: %d, hookid: %d", event, this->api_hookid);
     break;
-
-  // coverity[unterminated_default]
-  default:
-    ink_release_assert(false);
```
Comment (calavera) [inline proxy/ProxyClientSession.cc:165]: are you sure this is safe to remove?

Comment (calavera) [reply proxy/ProxyClientSession.cc:165]: I have no idea what it does or why it's here, tbh

Comment (mraerino) [reply proxy/ProxyClientSession.cc:165]: i basicly want to log all the cases. this is not really intended for production

**Comment by mraerino** [comment]
i'm working on backporting https://github.com/apache/trafficserver/pull/6364 to fix the underlying issue

**Comment by mraerino** [comment]
ok, this crashes in all kinds of ways. backporting seems to not have worked ☹️

**Pull request closed without merging**